keep-ui/app/providers/provider-form.tsx

```diff
@@ -151,86 +175,62 @@ const ProviderForm = ({
   console.log("Loading the ProviderForm component");
   const { mutate } = useProviders();
   const searchParams = useSearchParams();
-  const [activeTabsState, setActiveTabsState] = useState({});
-  const initialData = {
-    provider_id: provider.id, // Include the provider ID in formValues
-    ...formData,
-  };
-  if (provider.can_setup_webhook) {
-    initialData["install_webhook"] = provider.can_setup_webhook;
-    if (provider.pulling_enabled) {
-      initialData["pulling_enabled"] = true;
-    }
-  }
-  const [formValues, setFormValues] = useState<{
-    [key: string]: string | boolean;
-  }>(() => {
-    const initialValues = {
-      provider_id: provider.id,
-      ...formData,
-    };
-
-    // Set default values for select inputs
-    Object.entries(provider.config).forEach(([configKey, method]) => {
-      if (
-        method.type === "select" &&
-        method.default &&
-        !initialValues[configKey]
-      ) {
-        initialValues[configKey] = method.default;
-      }
-    });
-
-    if (provider.can_setup_webhook) {
-      initialValues["install_webhook"] = provider.can_setup_webhook;
-      if (provider.pulling_enabled) {
-        initialValues["pulling_enabled"] = provider.pulling_enabled;
-      }
-    }
-    return initialValues;
-  });
-
-  const [formErrors, setFormErrors] = useState<string>(null);
-  const [inputErrors, setInputErrors] = useState<{ [key: string]: boolean }>(
-    {}
+  const [formValues, setFormValues] = useState<ProviderFormData>(() =>
+    getInitialFormValues(provider)
```

Comment:
@coderabbitai this component has received several updates since this conversation. Currently, the `accessToken` variable has been taken out completely and its implementation encapsulated in an APIClient module.